cloudsecrets/aws.py

```diff
@@ -103,33 +114,43 @@ def _load_secrets(self) -> None:
         secrets = {}
         if self.create_if_not_present and not self._secret_exists:
             self._create_secret_resource()
-
-        params = dict(SecretId=self.secret)
-        if self._version:
-            params["VersionId"] = self._version
-
         try:
-            x = self.connection.get_secret_value(**params)
+            if self._version:
+                x = self.connection.get_secret_value(
+                    SecretId=self.secret, VersionId=self._version,
+                )
+            else:
+                x = self.connection.get_secret_value(SecretId=self.secret)
         except:
             self._encoded_secrets = {}
             self._secrets = {}
             return
+        kind = Secrets.unpack_response(x)
         self._version = x["VersionId"]
-        payload = x["SecretBinary"]
-        self._encoded_secrets = json.loads(payload)
-        for k, v in self._encoded_secrets.items():
-            secrets[k] = base64.b64decode(v).decode("ascii")
-        self._secrets = secrets
+        if self.is_binary:
+            payload = x["SecretBinary"]
+            self._encoded_secrets = json.loads(payload)
+            for k, v in self._encoded_secrets.items():
+                secrets[k] = base64.b64decode(v).decode("ascii")
+            self._secrets = secrets
+        else:
+            payload = x["SecretString"]
+            self._secrets = json.loads(payload)
 
     def _create_secret_resource(self) -> None:
         """
         Create the secret resource which will hold versions of secrets. A secret resource on its own has no secret data.
         """
         logging.debug(f"AWS _create_secret_resource")
         try:
-            self.connection.create_secret(
-                Name=self.secret, SecretBinary="{}".encode("UTF-8")
-            )
+            if self.is_binary:
+                self.connection.create_secret(
+                    Name=self.secret, SecretBinary="{}".encode("UTF-8")
+                )
+            else:
+                self.connection.create_secret(
+                    Name=self.secret, SecretString="{}".encode("UTF-8")
```

Comment:
@bowlofstew I don't think you can use utf-8 encoding here if you're storing in SecretString

Reply:
Can confirm this would generate 
```
Invalid type for parameter SecretString, value: b'{}', type: <class 'bytes'>, valid types: <class 'str'>
```

Reply:
On it now sir!